### DISCUSSION TITLE
Replace placeholder image with News, Weather or Home Assistant powered-feeds

### DISCUSSION THREAD

**Discussion by aniceberg** [OpenSurv/OpenSurv#16] [Feature Requests]
It would be a nice to use the static placeholder image to display a periodically-refreshing image/URL that rotates between news, weather forecasts, or some other card view (Home Assistant-powered in my case) .

Anyone have pointers of how this can be accomplished? I imagine I would create a HA view and put weather/news/status cards if I can display a URL natively which I'd hope will refresh automatically as the cards did periodically. If I'm limited to images, how do I generate an image every 5 or 10 seconds of that URL and then have opensurv refresh the current placeholder image?

**Comment by SvenVD** [comment]
This is in fact on my todo list. I want to have a third type of stream in OpenSurv which is opening an URL via a browser. I can't promise any timing currently, probably somewhere next year. In the mean time you can have another script generating an image of a website locally or published on an URL which you point OpenSurv to, OpenSurv will refresh it when image changes. I guess the image can be generated with something like https://playwright.dev/

**Reply by aniceberg** [reply]
> This is in fact on my todo list. I want to have a third type of stream in OpenSurv which is opening an URL via a browser. I can't promise any timing currently, probably somewhere next year. In the mean time you can have another script generating an image of a website locally or published on an URL which you point OpenSurv to, OpenSurv will refresh it when image changes. I guess the image can be generated with something like https://playwright.dev/

I was able to accomplish by adding the **Lovelace Kindle Screensaver** repo to Home Assistant: [https://github.com/sibbl/hass-lovelace-kindle-screensaver](https://github.com/sibbl/hass-lovelace-kindle-screensaver) and running it as an Add-On container (Settings-> Add-ons).

Follow the instructions, and you'll soon have a webserver running as a container on the same system as your Home Assistant is on that will serve the image based on the whatever settings you put in the container's configuration page.
<img width="636" alt="image" src="https://github.com/user-attachments/assets/dab66fbd-184b-48f1-b47c-05665de548d6" />


Then SSH into your OpenSurv system, and edit the **/etc/opensurv/monitor1.yml** file and add the container's URL to your streams like this:

>         - url: "http://homeassistant.local:5000/"
>           imageurl: true

And tada! Now I'm just trying to figure out if it's possible to hide or collapse Home Assistant's sidebar so it's doesn't take up a quarter of the rendered image.

**Reply by aniceberg** [reply]
***Update***
Ok, solved that too-- just append **?kiosk** to the HA_SCREENSHOT_URL in the configuration and make sure to install [Kiosk Mode](https://github.com/NemesisRE/kiosk-mode).

**Reply by SvenVD** [reply]
Very inventive!

**Reply by tuupos** [reply]
> > This is in fact on my todo list. I want to have a third type of stream in OpenSurv which is opening an URL via a browser. I can't promise any timing currently, probably somewhere next year. In the mean time you can have another script generating an image of a website locally or published on an URL which you point OpenSurv to, OpenSurv will refresh it when image changes. I guess the image can be generated with something like https://playwright.dev/
> 
> I was able to accomplish by adding the **Lovelace Kindle Screensaver** repo to Home Assistant: https://github.com/sibbl/hass-lovelace-kindle-screensaver and running it as an Add-On container (Settings-> Add-ons).
> 
> Follow the instructions, and you'll soon have a webserver running as a container on the same system as your Home Assistant is on that will serve the image based on the whatever settings you put in the container's configuration page. <img alt="image" width="636" src="https://private-user-images.githubusercontent.com/646958/411802860-dab66fbd-184b-48f1-b47c-05665de548d6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI0MjkwNDYsIm5iZiI6MTc1MjQyODc0NiwicGF0aCI6Ii82NDY5NTgvNDExODAyODYwLWRhYjY2ZmJkLTE4NGItNDhmMS1iNDdjLTA1NjY1ZGU1NDhkNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcxM1QxNzQ1NDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zZWMyNDY1NjYzZmVlM2RhNmNjOWJjNzk5OTY0MWNkM2RkZWU4YzZmMTM5YWU1MTcxNmNhZWU0MzRmZjVhMGFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Z76EbpdXxvEou7MN7mJT27HHJ3KqM74rBAPStDCcl6o">
> 
> Then SSH into your OpenSurv system, and edit the **/etc/opensurv/monitor1.yml** file and add the container's URL to your streams like this:
> 
> > ```
> >     - url: "http://homeassistant.local:5000/"
> >       imageurl: true
> > ```
> 
> And tada! Now I'm just trying to figure out if it's possible to hide or collapse Home Assistant's sidebar so it's doesn't take up a quarter of the rendered image.

This is awesome ! Works perfectly ! Thanks for this idea!

**Comment by tuupos** [comment]
That - would - be - awesome !

I'm searching for a way to put a flip clock on the screen:

https://github.com/PeterCang/flip-clock?form=MG0AV3
https://github.com/AlexanderLu98/Flip-Clock-Overlay-OBS-Studio?form=MG0AV3

![image](https://github.com/user-attachments/assets/18e0ae37-4a14-44d1-a6f7-5b4dbc9e6c1f)

This is now on a selfhosted server 192.x.x.x/index.html

**Comment by meilon** [comment]
For home assistant https://github.com/lanrat/hass-screenshot would be a great option to create a PNG of a dashboard
